Rework deb/rpm package build

JamesGuthrie opened the pull request:
## Description

- Use Rocky Linux to build rpms for RHEL 8 and 9

  Rocky is the new CentOS.

- Build deb for each distribution

  We've had issues with debian builds in the past, because we had the
  expectation that building on one version of debian would produce a
  shared lib which will execute on other debian-derivatives. Differences
  in glibc versions make this impossible (see https://github.com/timescale/promscale_extension/commit/68a1dab760cc0845659b8242f50595e1d23d6ba9).

  The above exposed differences in lintian, and an issue in the way that
  it was being invoked. The simplest fix is to disable erroring on
  warnings, which reduces the number of lints that we need to ignore.

- Add information about when each distribution will be EOL

  I keep on forgetting, so having a link to the details helps.

- Rework release-tester make target

  This is a slightly experimental approach. There is an inherent
  mismatch between make's file-oriented approach, and docker's "image
  stored in daemon" approach. The intermediate steps have been reworked
  to produce a file artifact, which should prevent having to rebuild
  things all of the time.

  This change prompted the removal of *.deb and *.rpm from .dockerignore
  as those files are required to build the release-tester make target.

## Merge requirements

Please take into account the following non-code changes that you may need to make with your PR:

- [ ] ~~CHANGELOG entry for user-facing changes~~
- [ ] ~~Updated the relevant documentation~~

JamesGuthrie commented on .github/workflows/release.yaml:
```diff
@@ -115,29 +142,25 @@ jobs:
-            . ; \
-          ./tools/smoke-test "${EXTENSION_DOCKER_IMAGE}" "${DOCKER_PLATFORM}"; \
-        done
+        EXTENSION_DOCKER_IMAGE="${IMAGE_BASE}-${DISTRO}-${DISTRO_VERSION}"; \
```
Hmmm... Indeed. Good catch!

sumerman reviewed (changes requested):
I suspect the pgtap stuff got into this PR by accident. Otherwise LGTM.

JamesGuthrie commented:
> I suspect the pgtap stuff got into this PR by accident. Otherwise LGTM.

🙈  well that was me being sloppy with branches... Fixed. I've now squashed the fixes down into one commit.
